dashboard/app/models/pd/workshop.rb

```diff
@@ -580,6 +580,34 @@ def attending_teachers
     sessions.flat_map(&:attendances).flat_map(&:teacher).uniq
   end
 
+  # Get all teachers who have attended all sessions of this workshop.
+  def teachers_attending_all_sessions(cdo_scholarship=false)
```

Comment:
This whole method might be a good candidate for a [Query object](https://github.com/code-dot-org/code-dot-org/tree/staging/dashboard/lib/queries) (read linked doc and ask Elijah for details).